saw-core/src/Verifier/SAW/Simulator/Value.hs

```diff
@@ -49,22 +50,27 @@ Values are parameterized by the /name/ of an instantiation.
 The concrete parameters to use are computed from the name using
 a collection of type families (e.g., 'EvalM', 'VBool', etc.). -}
 data Value l
-  = VFun !(Thunk l -> MValue l)
+  = VFun !LocalName !(Thunk l -> MValue l)
   | VUnit
   | VPair (Thunk l) (Thunk l) -- TODO: should second component be strict?
-  | VCtorApp !Ident !(Vector (Thunk l))
+  | VCtorApp !(PrimName (TValue l)) ![Thunk l] ![Thunk l]
   | VVector !(Vector (Thunk l))
   | VBool (VBool l)
   | VWord (VWord l)
-  | VToNat (Value l)
+  | VBVToNat !Int (Value l) -- TODO: don't use @Int@ for this, use @Natural@
```

Comment:
Is there anything tricky about changing this to `Natural`?

Reply:
Mainly that vectors and the concrete bitvector representations use `Int` for their lengths.